configuration/driver/icedrv_init_column.F90

```diff
@@ -218,27 +219,34 @@ subroutine init_shortwave
 
          enddo
 
-         do i = 1, nx
-
-            if (trim(shortwave(1:4)) == 'dEdd') then ! delta Eddington
+         if (trim(shortwave(1:4)) == 'dEdd') then ! delta Eddington
+            ! initialize orbital parameters
+            ! These come from the driver in the coupled model.
+            call icepack_warnings_flush(nu_diag)
+            call icepack_init_orbit()
+            call icepack_warnings_flush(nu_diag)
+            if (icepack_warnings_aborted()) &
+               call icedrv_system_abort(i, istep1, subname, __FILE__, __LINE__)
 
-               ! initialize orbital parameters
-               ! These come from the driver in the coupled model.
-               call icepack_warnings_flush(nu_diag)
-               call icepack_init_orbit()
+            if (trim(shortwave) == 'dEdd_snicar') then
+               use_snicar = .true. ! 5-band SNICAR scheme for snow cover
+               call icepack_init_parameters(use_snicar_in=use_snicar, snw_ssp_table_in=snw_ssp_table)
                call icepack_warnings_flush(nu_diag)
-               if (icepack_warnings_aborted()) &
-                  call icedrv_system_abort(i, istep1, subname, __FILE__, __LINE__)
-
-               if (trim(shortwave) == 'dEdd_snicar') then
-                  use_snicar = .true. ! 5-band SNICAR scheme for snow cover
-                  call icepack_init_parameters(use_snicar_in=use_snicar)
-                  call icepack_warnings_flush(nu_diag)
-                  if (icepack_warnings_aborted()) call icedrv_system_abort(string=subname, &
-                      file=__FILE__,line= __LINE__)
+               if (icepack_warnings_aborted()) call icedrv_system_abort(string=subname, &
+                   file=__FILE__,line= __LINE__)
+               if (trim(snw_ssp_table) /= 'test') then
+                  call init_snicarssptable()
                endif
+            endif
+         endif ! dEdd
 
-            endif ! dEdd
+         call icepack_init_radiation()
```

Comment:
so far icepack_init_radiation is only needed for dEdd, so this sectioncould be called inside the endif statement immediately above

Reply:
icepack_init_radiation calls the data_dEdd_3band which is called all the time right now.  My preference would be that the icepack_init_radiation call be always called and then the if tests be inside that subroutine.  That provides extensibility in the future for the icepack_init_radiation call if other stuff needs to be initialized for non-dEdd options.   What I'll do is add an "if dEdd" around the data_dEdd_3band call in the subroutine and check that doesn't break anything.